dimagi/commcare-hq
# Alt text for case list icons

## Product Description
Adds an textbox to enter alt text on the key value mapping when the values are icons. For case list icons/clickable icons, this value is used in web apps to set the alt text on the icon, and for clickable icons is updated when the icon changes.
<img width="583" alt="image" src="https://github.com/dimagi/commcare-hq/assets/100609770/86ac5f30-d1c9-4819-b5aa-da3ec96faf32">

## Technical Summary
[USH-3933](https://dimagi.atlassian.net/browse/USH-3933)
https://github.com/dimagi/formplayer/pull/1536
https://github.com/dimagi/commcare-core/pull/1399, https://github.com/dimagi/commcare-core/pull/1400
Adds `alt_text` as a `LabelProperty` (`{'lang': 'value'}`) to `MappingItem`. UI is added in case list config to save this property. The property is used to generate an `alt_text` node in the suite for case list icons and clickable icons that lives within the `field` element, using the same calculation and variable names as its corresponding `value`, e.g.:
```xml
<alt_text>
  <text>
    <xpath function="if(starred = '0', $k0, if(starred = '1', $k1, ''))">
      <variable name="k0">
        <locale id="m0.case_short.case_starred_1.alt_text.k0"/>
      </variable>
      <variable name="k1">
        <locale id="m0.case_short.case_starred_1.alt_text.k1"/>
      </variable>
    </xpath>
  </text>
</alt_text>
```

## Feature Flag
Clickable icons behind CASE_LIST_CLICKABLE_ICON.

## Safety Assurance

### Safety story
Alt text is only added to the suite for appropriate column types "enum-image" and "clickable-icon" and is an optional element in Formplayer. Tested locally with configs for clickable icons and regular case list icons, including alternate language alt text.

### Automated test coverage
Automated tests added for alt text in suite and app strings. Additionally updated tests where `alt_text` is now an expected node in the suite, as with clickable icons.

### QA Plan
[QA-6080](https://dimagi.atlassian.net/browse/QA-6080)
This will get QA to ensure alt text is correctly saved and ultimately used by web apps.

### Rollback instructions
This should be deployed with or after Formplayer changes, otherwise apps with clickable icons or case list icons will not load correctly.

This PR can be reverted independently, but if Formplayer changes need to be reverted, this PR should be as well or it will break apps with now-invalid suite files.

### Labels & Review


[USH-3933]: https://dimagi.atlassian.net/browse/USH-3933?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[QA-6080]: https://dimagi.atlassian.net/browse/QA-6080?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
